src/cmd/build.js

```diff
@@ -37,14 +39,56 @@ export type PackageCreatorParams = {
   artifactsDir: string,
 };
 
+export type LocalizedNameParams = {
+  messageFile: string,
+  manifestData: ExtensionManifest,
+}
+
 export type PackageCreatorFn =
     (params: PackageCreatorParams) => Promise<ExtensionBuildResult>;
 
+export async function getDefaultLocalizedName(
+  {messageFile, manifestData}: LocalizedNameParams
+): Promise<string> {
+
+  let messageData: any;
+  let extensionName: string = manifestData.name;
+
+  try {
+    messageData = await fs.readFile(messageFile);
+  }
+  catch (err) {
+    log.info(err);
+    throw new WebExtError(`${err}`);
+  }
+
+  try {
+    messageData = JSON.parse(messageData);
+  }
+  catch (err) {
+    throw new WebExtError(`The JSON file is malformed: ${err}`);
+  }
+  extensionName = manifestData.name.replace(/__MSG_([A-Za-z0-9@_]+?)__/g,
+                      (match, messageName) => {
+                        log.info(messageData[messageName]);
+                        if (!(messageData[messageName]
+                            && messageData[messageName].message)) {
+                          let error = new WebExtError
+                        ('The locale file does not have the correct format');
```

Comment:
This should give the developer a clue as to which key was missing by providing the key value in the error message. Something like ``The locale file ${messageFile} is missing a message for key: ${messageName}``